src/main/java/io/confluent/copycat/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -25,41 +25,86 @@
 
   public static final String HDFS_URL_CONFIG = "hdfs.url";
   private static final String HDFS_URL_DOC = "HDFS connection URL.";
+
   public static final String RECORD_WRITER_PROVIDER_CLASS_CONFIG = "record.writer.provider.class";
   private static final String RECORD_WRITER_PROVIDER_CLASS_DOC = "";
+
   public static final String FLUSH_SIZE_CONFIG = "flush.size";
   private static final String FLUSH_SIZE_DOC = "The number of records needed to rotate files.";
+
   public static final String ROTATE_INTERVAL_CONFIG = "rotate.interval.ms";
   private static final String ROTATE_INTERVAL_DOC = "The interval to rotate files.";
-  public static final String TOPIC_DIR_CONFIG = "topic.dir";
-  private static final String TOPIC_DIR_DOC = "HDFS directory to store data.";
-  public static final String TOPIC_DIR_DEFAULT = "topics";
+
+  public static final String TOPICS_DIR_CONFIG = "topics.dir";
+  private static final String TOPICS_DIR_DOC = "HDFS directory to store data.";
+  public static final String TOPICS_DIR_DEFAULT = "topics";
+
+  public static final String LOGS_DIR_CONFIG = "logs.dir";
+  private static final String LOGS_DIR_DOC = "HDFS directory to store logs.";
+  public static final String LOGS_DIR_DEFAULT = "logs";
+
   public static final String SCHEMA_CACHE_SIZE_CONFIG = "schema.cache.size";
-  public static final int SCHEMA_CACHE_SIZE_DEFAULT = 1000;
   private static final String SCHEMA_CACHE_SIZE_DOC = "The size of the schema cache.";
+  public static final int SCHEMA_CACHE_SIZE_DEFAULT = 1000;
+
   public static final String STORAGE_CLASS_CONFIG = "storage.class";
   private static final String STORAGE_CLASS_DOC = "The storage layer.";
   public static final String STORAGE_CLASS_DEFAULT = "io.confluent.copycat.hdfs.HdfsStorage";
+
   public static final String RETRY_BACKOFF_CONFIG = "retry.backoff.ms";
   private static final String RETRY_BACKOFF_DOC = "The retry backoff in milliseconds.";
   public static final long RETRY_BACKOFF_DEFALUT = 5000L;
+
   public static final String HIVE_HOME_CONFIG = "hive.home";
   private static final String HIVE_HOME_DOC = "Hive home directory";
   public static final String HIVE_CONF_DIR_CONFIG = "hive.conf.dir";
   private static final String HIVE_CONF_DIR_DOC = "Hive configuration directory";
 
+  public static final String PARTITIONER_CLASS_CONFIG = "partitioner.class";
+  private static final String PARTITIONER_CLASS_DOC = "The partitioner class.";
+  public static final String PARTITIONER_CLASS_DEFALUT = "io.confluent.copycat.hdfs.partitioner.DefaultPartitioner";
+
+  public static final String PARTITION_FIELD_CONFIG = "partition.field";
```

Comment:
Not a problem for this round, but something to think about refactoring in the future -- you can separate out configs for different classes into their own AbstractConfig subclasses. That's one way to get around a) having all the configs in one place and b) having to manually validate fields that are required for only some of the partitioners. Of course, the drawback to not having all the configs in one class is that automatically generating docs will be a bit harder.